state/documentdb.yaml

```diff
@@ -0,0 +1,171 @@
+  BackupRetentionPeriod:
+    Description: 'The number of days for which automated backups are retained.'
+    Type: Number
+    MinValue: 1
+    MaxValue: 35
+    Default: 30
+  DBClusterParameterGroupName:
+    Description: 'Optional name of an existing DB parameter group.'
+    Type: String
+    Default: ''
+  MasterUsername:
+    Description: 'The name of the master user for the cluster.'
+    Type: String
+    MinLength: 1
+    MaxLength: 63
+    Default: master
+  MasterUserPassword:
+    Description: 'The password for the master database user. This password can contain any printable ASCII character except forward slash (/), double quote ("), or the "at" symbol (@).'
+    Type: String
+    MinLength: 8
+    MaxLength: 100
+    NoEcho: true
+  PreferredBackupWindow:
+    Description: 'The daily time range (in UTC) during which automated backups are created.'
+    Type: String
+    Default: '09:54-10:24'
+  PreferredMaintenanceWindow:
+    Description: 'The weekly time range (in UTC) during which system maintenance can occur (must not conflict with the backup window).'
+    Type: String
+    Default: 'sat:07:00-sat:07:30'
+  SnapshotIdentifier:
+    Description: 'Optional identifier for cluster snapshot (ARN) to restore from.'
+    Type: String
+    Default: ''
+  DBInstanceClass:
+    Description: 'The compute and memory capacity of the instance.'
+    Type: String
+    Default: 'db.r4.large'
+Conditions:
+  HasParentKmsKeyStack: !Not [!Equals [!Ref ParentKmsKeyStack, '']]
+  HasParentBastionStack: !Not [!Equals [!Ref ParentBastionStack, '']]
+  HasDBClusterParameterGroupName: !Not [!Equals [!Ref DBClusterParameterGroupName, '']]
+  HasSnapshotIdentifier: !Not [!Equals [!Ref SnapshotIdentifier, '']]
+Resources:
+  DBSubnetGroup:
+    Type: 'AWS::DocDB::DBSubnetGroup'
+    Properties:
+      DBSubnetGroupDescription: !Ref 'AWS::StackName'
+      SubnetIds: !Split [',', {'Fn::ImportValue': !Sub '${ParentVPCStack}-SubnetsPrivate'}]
+  SecurityGroup:
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      GroupDescription: !Ref 'AWS::StackName'
+      VpcId: {'Fn::ImportValue': !Sub '${ParentVPCStack}-VPC'}
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 27017
+        ToPort: 27017
+        SourceSecurityGroupId: {'Fn::ImportValue': !Sub '${ParentClientStack}-ClientSecurityGroup'}
+  SecurityGroupInBastion:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Condition: HasParentBastionStack
+    Properties:
+      GroupId: !Ref SecurityGroup
+      IpProtocol: tcp
+      FromPort: 27017
+      ToPort: 27017
+      SourceSecurityGroupId: {'Fn::ImportValue': !Sub '${ParentBastionStack}-SecurityGroup'}
+  Cluster: # FIXME add monitoring
+    # TODO not yet supported by CloudFormation DeletionPolicy: Snapshot
+    # TODO not yet supported by CloudFormation UpdateReplacePolicy: Snapshot
+    Type: 'AWS::DocDB::DBCluster'
+    Properties:
+      AvailabilityZones:
+      - {'Fn::ImportValue': !Sub '${ParentVPCStack}-AZA'}
```

Comment:
What's the impact of deploying to two AZs only?

Reply:
I improved the template to use all available AZs